unity-client/Assets/UnityGLTF/Scripts/Loader/WebRequestLoader.cs

```diff
@@ -66,53 +60,31 @@ private IEnumerator CreateHTTPRequest(string rootUri, string httpRequestPath)
             {
                 finalUrl = Path.Combine(rootUri, httpRequestPath);
             }
-            
-            UnityWebRequest www = null; 
-            bool retry = true;
-            
-            while (retry)
-            {
-                retry = false;
-                www = new UnityWebRequest(finalUrl, "GET", new DownloadHandlerBuffer(), null);
 
-                www.timeout = 5000;
-#if UNITY_2017_2_OR_NEWER
-                yield return www.SendWebRequest();
-#else
-            yield return www.Send();
-#endif
-                if ((int)www.responseCode >= 400)
+            return DCL.WebRequestController.i.Get(
+                url: finalUrl,
+                downloadHandler: new DownloadHandlerBuffer(),
+                OnSuccess: (webRequestResult) =>
                 {
-                    Debug.LogError($"{www.responseCode} - {www.url}");
-                    
-                    //Note (Adrian): 500 to 600 codes are reserved codes to Server error responses, so if we have an error from server, we retry
-                    if (retryCont < RETRY_AMOUNTS && www.responseCode >= 500 &&  www.responseCode < 600)
+                    if (webRequestResult.downloadedBytes > int.MaxValue)
                     {
-                        retryCont++;
-                        retry = true;
+                        Debug.LogError("Stream is too big for a byte array");
                     }
-                    else
+                    else if (webRequestResult.downloadHandler.data != null)
```

Comment:
This creates a whole copy of all the data of the request, we shouldn't call it twice. Please check the comment below.

Reply:
You're right! I will adapt it, thanks!

Reply:
Done in https://github.com/decentraland/explorer/pull/2256/commits/810b7caaa2a8c27b21c55bed7ca5c6768dd00aad